src/lexer.c

```diff
@@ -508,6 +508,75 @@ void tryParseObjectDefineOrKeys (bool keys) {
         else break;
 
 
+        // `if (Object.prototype.hasOwnProperty.call(`  IDENTIFIER `, ` IDENTIFIER$2 `)) return` `;`?
+        if (ch == 'i' && *(pos + 1) == 'f') {
+          uint16_t *ifStartPos = pos;
+
+          pos += 2;
+          ch = commentWhitespace();
+          if (ch != '(') break;
+          pos++;
+          ch = commentWhitespace();
+          if (ch != 'O' || !str_eq5(pos + 1, 'b', 'j', 'e', 'c', 't')) {
+            // Revert parsing the current optional if statement, but don't bail
+            // out since we can try parse the next possible if statement.
+            pos = ifStartPos;
+            ch = 'i';
+            goto currentIfStatementEnd;
+          }
+          pos += 6;
+          ch = commentWhitespace();
+          if (ch != '.') {
+            // Revert parsing the current optional if statement, but don't bail
+            // out since we can try parse the next possible if statement.
+            pos = ifStartPos;
+            ch = 'i';
+            goto currentIfStatementEnd;
```

Comment:
How do you feel about this usage of `goto` (and labeled break in JS)? It's to quit parsing this `if` statement, while still checking the next possible `if`.

Reply:
Dijkstra won't mind.